specifyweb/backend/businessrules/utils.py

```
@@ -1,33 +1,230 @@
             if not isinstance(unique_catnum_enabled, bool):
                 unique_catnum_enabled = False
 
     except json.JSONDecodeError:
-        logger.warning(f"Error: Could not decode JSON for collection preferences")
+        logger.warning(
+            f"Error: Could not decode JSON for collection preferences")
     except TypeError as e:
-        logger.warning(f"Error: Unexpected data structure in collection preferences: {e}")
+        logger.warning(
+            f"Error: Unexpected data structure in collection preferences: {e}")
     except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)